src/App/DocumentObject.cpp

```diff
@@ -292,36 +292,48 @@ std::string DocumentObject::getFullLabel() const {
     return name;
 }
 
-const char *DocumentObject::getNameInDocument() const
+const char* DocumentObject::getDagKey() const
+{
+    if(!pcNameInDocument)
+    {
+        return nullptr;

```

Comment:
And what's better compared to getNameInDocument() when it returns a null pointer here?

Reply:
It is better because when people call `getNameInDocument()` they should get a name (as most of FreeCAD code actually expects).

Now, the DAG problem is a different one I am not much aware. Do the "DAG people" want a null pointer? I don't know... it is up for them to decide. If the very specific group of people interested in the DAG key decide to forbid a null pointer, then can make a PR. :-)